addons/mrp/models/mrp_bom.py

```diff
@@ -138,6 +137,14 @@ def _bom_find(self, product_tmpl=None, product=None, picking_type=None, company_
         if bom_type:
             domain += [('type', '=', bom_type)]
         # order to prioritize bom with product_id over the one without
+        return domain
+
+    @api.model
+    def _bom_find(self, product_tmpl=None, product=None, picking_type=None, company_id=False, bom_type=False):
+        """ Finds BoM for particular product, picking and company """
+        domain = self._bom_find_domain(product_tmpl=product_tmpl, product=product, picking_type=picking_type, company_id=company_id, bom_type=bom_type)
+        if domain is False:
```

Comment:
maybe _bom_find_domain could raise an error if both kwargs product and product_tmpl are False, so it does not need to be manage in function that use it. Bom find is probably never used without a product specified.